Sources/XibLoc/Plurality/PluralValue.swift

```diff
@@ -17,37 +17,264 @@ import Foundation
+	}
+	
+	public init(float: Float, format: NumberFormat) {
+		/* Doc of `NSString` for init with format and locale says if we specify a
+		 * `nil` locale, we get the system locale. The system locale, says the
+		 * doc, is the locale to use when we don’t want any localizations, which
+		 * is exactly what we want!
+		 * The format `%.*f` will take the float it is given and output exactly
+		 * the given number fraction digits.
+		 * The `%f` format work for floats and doubles. So as long as Swift does
+		 * not represent floats or doubles using `long double` internally, we will
+		 * be good passing them to String(format:). */
+		let stringValue = String(format: "%.*f", locale: nil, float, format.maxFractionDigits)
+		let components = stringValue.split(separator: ".", omittingEmptySubsequences: false)
+		assert(components.count == 2 || components.count == 1)
+		
+		let intPart = String(components[0])
+		var fractionPart = String(components.count >= 2 ? components[1] : "")
+		assert(fractionPart.count == format.maxFractionDigits)
+		while fractionPart.hasSuffix("0") && fractionPart.count > format.minFractionDigits {
+			fractionPart.removeLast()
 		}
 		
+		self.init(intPart: intPart, fractionPart: fractionPart, forceDecimalSeparator: format.forceDecimalSeparator, forceNegativeZero: format.zeroIsNegative)!
+	}
+	
+	public init?(string: String) {
+		let components = string.split(separator: ".", omittingEmptySubsequences: false)
+		guard components.count == 1 || components.count == 2 else {return nil}
+		self.init(
+			intPart: String(components[0]),
+			fractionPart: components.count >= 2 ? String(components[1]) : nil
+		)
 	}
 	
-	case int(Int)
-	case float(Float)
-	case floatCustomCharacteristics(value: Float, characteristics: FloatCharacteristics)
+	public init?(intPart i: String, fractionPart f: String?) {
+		let negative = (i.first == "-")
+		let i = (negative ? String(i.dropFirst()) : i)
+		self.init(
+			intPart: i,
+			fractionPart: f,
+			isNegative: negative
+		)
+	}
+	
+	/** Init a PluralValue directly with its parts.
 	
-	public func asNumber() -> NSNumber {
-		switch self {
-			case .int(let i):                           return NSNumber(value: i)
-			case .float(let f):                         return NSNumber(value: f)
-			case .floatCustomCharacteristics(let f, _): return NSNumber(value: f)
+	Init fails if either:
+	- The int or fraction part contains a non-numeric (base-10) character;
+	- The int part has a 0 prefix (and is not equal to 0);
+	- The int part is empty. */
+	public init?(intPart i: String, fractionPart f: String?, isNegative n: Bool) {
+		guard
+			!i.isEmpty,
+			(!i.hasPrefix("0") || i == "0"),
+			i.rangeOfCharacter(from: CharacterSet(charactersIn: "0123456789").inverted, options: [.literal]) == nil,
+			f?.rangeOfCharacter(from: CharacterSet(charactersIn: "0123456789").inverted, options: [.literal]) == nil
```

Comment:
Nope. I wasn’t sure, but your comment made me check, and indeed I can’t, because `decimalDigits` includes digits in all localisations and we only want latin digits. The doc says:
> These characters include, for example, the decimal digits of the Indic scripts and Arabic.